src/gradualizer.erl

```diff
@@ -171,13 +171,18 @@ type_check_file(File, Opts) ->
                     Includes = proplists:get_all_values(i, Opts),
                     case gradualizer_file_utils:get_forms_from_erl(File, Includes) of
                         {ok, Forms} ->
+                            % import the currently checked file so that it can
+                            % reference itself even without adding its path via --pa
+                            ok = gradualizer_db:import_erl_files([File]),
```

Comment:
@xxdavid See @zuiderkwast's [comment about this](https://github.com/josefs/Gradualizer/pull/516#discussion_r1124883427) - he might be right that this should be pushed even deeper in the case of `.erl` files.

Reply:
Even deeper, i.e. practically just before `lint_and_check_forms` calls `type_check_forms`.

Reply:
Good catch! I've moved it a bit deeper.